[Bug] sqlservercdc throw a NullPointerException when table name contains point

### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.


### Flink version

1.13.6

### Flink CDC version

2.5-SNAPSHOT

### Database and its version

sqlserver 2016

### Minimal reproduce step

`SqlServerSourceBuilder.SqlServerIncrementalSource<String> sqlServerSource =
                new SqlServerSourceBuilder()
                        .hostname("10.10.17.35")
                        .port(1433)
                        .databaseList("CRDB")
                        .tableList("dbo.tCR0021_V2.0")
                        .username("app")
                        .password("app127")
                        .deserializer(new JsonDebeziumDeserializationSchema())
                        .startupOptions(StartupOptions.initial())
                        .build();
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        // enable checkpoint
        // set the source parallelism to 2

        env.fromSource(
                sqlServerSource,
                WatermarkStrategy.noWatermarks(),
                "SqlServerIncrementalSource")
                .setParallelism(2)
                .print()
                .setParallelism(1);

        env.execute("Print SqlServer Snapshot + Change Stream");`



### What did you expect to see?

正常执行

### What did you see instead?

Executing this code throws a NullPointerException

Caused by: java.lang.RuntimeException: SplitFetcher thread 0 received unexpected exception while polling the records
	at org.apache.flink.connector.base.source.reader.fetcher.SplitFetcher.runOnce(SplitFetcher.java:150)
	at org.apache.flink.connector.base.source.reader.fetcher.SplitFetcher.run(SplitFetcher.java:105)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
Caused by: org.apache.flink.util.FlinkRuntimeException: Read split SnapshotSplit{tableId=CRDB.dbo.tCR0021_V2, splitId='CRDB.dbo.tCR0021_V2.0:1', splitKeyType=[`ID` INT NOT NULL], splitStart=[9084], splitEnd=[18167], highWatermark=null} error due to java.lang.NullPointerException.
	at com.ververica.cdc.connectors.base.source.reader.external.IncrementalSourceScanFetcher.checkReadException(IncrementalSourceScanFetcher.java:182)
	at com.ververica.cdc.connectors.base.source.reader.external.IncrementalSourceScanFetcher.pollSplitRecords(IncrementalSourceScanFetcher.java:129)
	at com.ververica.cdc.connectors.base.source.reader.IncrementalSourceSplitReader.fetch(IncrementalSourceSplitReader.java:73)
	at org.apache.flink.connector.base.source.reader.fetcher.FetchTask.run(FetchTask.java:58)
	at org.apache.flink.connector.base.source.reader.fetcher.SplitFetcher.runOnce(SplitFetcher.java:142)
	... 7 more
Caused by: io.debezium.DebeziumException: java.lang.NullPointerException
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask$SqlServerSnapshotSplitReadTask.execute(SqlServerScanFetchTask.java:262)
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask.execute(SqlServerScanFetchTask.java:94)
	at com.ververica.cdc.connectors.base.source.reader.external.IncrementalSourceScanFetcher.lambda$submitTask$0(IncrementalSourceScanFetcher.java:95)
	... 6 more
Caused by: java.lang.NullPointerException
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask$SqlServerSnapshotSplitReadTask.createDataEventsForTable(SqlServerScanFetchTask.java:335)
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask$SqlServerSnapshotSplitReadTask.createDataEvents(SqlServerScanFetchTask.java:319)
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask$SqlServerSnapshotSplitReadTask.doExecute(SqlServerScanFetchTask.java:289)
	at com.ververica.cdc.connectors.sqlserver.source.reader.fetch.SqlServerScanFetchTask$SqlServerSnapshotSplitReadTask.execute(SqlServerScanFetchTask.java:257)
	... 8 more


### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializer.java
+++ b/flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializer.java
@@ -70,7 +70,7 @@ public abstract class SourceSplitSerializer
             boolean useCatalogBeforeSchema =
                     SerializerUtils.shouldUseCatalogBeforeSchema(snapshotSplit.getTableId());
             out.writeBoolean(useCatalogBeforeSchema);
-            out.writeUTF(snapshotSplit.getTableId().toString());
+            out.writeUTF(snapshotSplit.getTableId().toDoubleQuotedString());
             out.writeUTF(snapshotSplit.splitId());
             out.writeUTF(snapshotSplit.getSplitKeyType().asSerializableString());
 

--- a/flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializerTest.java
+++ b/flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializerTest.java
@@ -29,8 +29,9 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.Assert.assertEquals;
 
+/** Tests for {@link SourceSplitSerializer}. */
 public class SourceSplitSerializerTest {
 
     @Test

--- a/flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializerTest.java
+++ b/flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/source/meta/split/SourceSplitSerializerTest.java
@@ -1,0 +1,96 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.base.source.meta.split;
+
+import org.apache.flink.table.types.logical.BigIntType;
+import org.apache.flink.table.types.logical.RowType;
+
+import com.ververica.cdc.connectors.base.source.meta.offset.Offset;
+import com.ververica.cdc.connectors.base.source.meta.offset.OffsetFactory;
+import io.debezium.relational.TableId;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SourceSplitSerializerTest {
+
+    @Test
+    public void testSnapshotTableIdSerializeAndDeserialize() throws IOException {
+        SnapshotSplit snapshotSplitBefore =
+                new SnapshotSplit(
+                        new TableId("cata`log\"", "s\"che`ma", "ta\"ble.1`"),
+                        "test",
+                        new RowType(
+                                Collections.singletonList(
+                                        new RowType.RowField("id", new BigIntType()))),
+                        null,
+                        null,
+                        null,
+                        new HashMap<>());
+
+        SourceSplitSerializer sourceSplitSerializer =
+                new SourceSplitSerializer() {
+                    @Override
+                    public OffsetFactory getOffsetFactory() {
+                        return new OffsetFactory() {
+                            @Override
+                            public Offset newOffset(Map<String, String> offset) {
+                                return null;
+                            }
+
+                            @Override
+                            public Offset newOffset(String filename, Long position) {
+                                return null;
+                            }
+
+                            @Override
+                            public Offset newOffset(Long position) {
+                                return null;
+                            }
+
+                            @Override
+                            public Offset createTimestampOffset(long timestampMillis) {
+                                return null;
+                            }
+
+                            @Override
+                            public Offset createInitialOffset() {
+                                return null;
+                            }
+
+                            @Override
+                            public Offset createNoStoppingOffset() {
+                                return null;
+                            }
+                        };
+                    }
+                };
+
+        SnapshotSplit snapshotSplitAfter =
+                (SnapshotSplit)
+                        sourceSplitSerializer.deserialize(
+                                sourceSplitSerializer.getVersion(),
+                                sourceSplitSerializer.serialize(snapshotSplitBefore));
+
+        assertEquals(snapshotSplitBefore.getTableId(), snapshotSplitAfter.getTableId());
+    }
+}

